api/censuses.go

```diff
@@ -529,18 +578,31 @@ func (a *API) censusProofHandler(msg *apirest.APIdata, ctx *httprouter.HTTPConte
 	if err != nil {
 		return err
 	}
-	keyHash, err := ref.Tree().Hash(key)
+
+	leafKey := key
+	if ref.CensusType != int32(models.Census_ARBO_POSEIDON) {
+		leafKey, err = ref.Tree().Hash(key)
+		if err != nil {
+			return err
+		}
+	}
+
+	leafV, siblings, err := ref.Tree().GenProof(leafKey)
 	if err != nil {
 		return err
 	}
-	leafV, siblings, err := ref.Tree().GenProof(keyHash)
+
+	// Get the leaf siblings from arbo based on the key received and include
+	// them into the response
+	circomSiblings, err := ref.Tree().GetCircomSiblings(leafKey)
```

Comment:
Shouldn't we only return the circom siblings if the census type is zkWeighted?

Reply:
Yes, I am agree